td-agent/Gemfile

```diff
@@ -50,24 +53,24 @@ gem "fluent-plugin-record-modifier", "2.1.0"
 gem "fluent-plugin-rewrite-tag-filter", "2.3.0"
 gem "nokogiri", "1.10.10"
 
-if /mswin|mingw/ =~ RUBY_PLATFORM
-  gem "ffi", "1.13.1"
-  gem "ffi-win32-extensions", "1.0.3"
-  gem "win32-api", "1.9.1"
-  gem "win32-event", "0.6.3"
-  gem "win32-ipc", "0.7.0"
-  gem "win32-service", "2.1.5"
-  gem "windows-api", "0.4.4"
-  gem "windows-pr", "1.2.6"
-  gem "winevt_c", "0.8.1"
-  gem "win32-eventlog", "0.6.7"
-  gem "fluent-plugin-parser-winevt_xml", "0.2.2"
-  gem "fluent-plugin-windows-eventlog", "0.7.0"
-else
-  gem "rdkafka", "0.8.0"
-  gem "systemd-journal", "1.3.3"
-  gem "fluent-plugin-systemd", "1.0.2"
-end
+windows_platforms = [:mingw, :x64_mingw] # :mswin
+gem "ffi", "1.13.1", platforms: windows_platforms
+gem "ffi-win32-extensions", "1.0.3", platforms: windows_platforms
+gem "win32-api", "1.9.1", platforms: windows_platforms
+gem "win32-event", "0.6.3", platforms: windows_platforms
+gem "win32-ipc", "0.7.0", platforms: windows_platforms
+gem "win32-service", "2.1.5", platforms: windows_platforms
+gem "windows-api", "0.4.4", platforms: windows_platforms
+gem "windows-pr", "1.2.6", platforms: windows_platforms
+gem "winevt_c", "0.8.1", platforms: windows_platforms
+gem "win32-eventlog", "0.6.7", platforms: windows_platforms
+gem "fluent-plugin-parser-winevt_xml", "0.2.2", platforms: windows_platforms
+gem "fluent-plugin-windows-eventlog", "0.7.0", platforms: windows_platforms
+
+not_windows_platforms = [:ruby]
+gem "rdkafka", "0.8.0", platforms: not_windows_platforms
+gem "systemd-journal", "1.3.3", platforms: not_windows_platforms
+gem "fluent-plugin-systemd", "1.0.2", platforms: not_windows_platforms
```

Comment:
nitpick: Systemd is Linux only. But macOS users seems to be a few.
It is not issued in the near future.